master/internal/kubernetes/tokens.go

```diff
@@ -0,0 +1,125 @@
+package kubernetes
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/pkg/actor/actors"
+)
+
+// The tokens actor is responsible for providing tokens to the pod actors which indicate
+// to the pod actors that they are allowed to make calls to the Kubernetes API server
+// to create Kubernetes resources (configMaps and pods).
+//
+// There are two reasons the token system is required as opposed to allowing the pod actors
+// to create Kubernetes resources asynchronously:
+//
+//    1) Each pod creation first requires the creation of a configMap, however creating the two
+//       is not an atomic operation. If there is a large number of concurrent creation requests
+//       (e.g., a large HP search experiment) the kubernetes API server ends up processing the
+//       creation of all the configMaps before starting to create pods, which adds significant
+//       latency to the creation of pods.
+//
+//    2) If all creation requests are submitted asynchronously, it is possible the Kubernetes API
+//       server will temporarily become saturated, and be slower to respond to other requests.
+//
+//  The reason that we implemented a token system rather than creating a worker pool which
```

Comment:
blocking: I 100% like those comments to explain the design choice. Can we add one comment to explain the message protocol? It might seem obvious that we go with `requestToken`, `grantToken`, and `releaseToken` but a comment would be helpful to understand.